### PR TITLE
[op-conductor] Implement raft consensus

### PR DESCRIPTION
**Pull request by 0x00101010**
**Description**

Define and implement consensus interface for sequencer leader election.

**Tests**

Unit tests added for CommitUnsafePayload & LatestUnsafePayload, did not test other functions as they're purely proxy functions to Raft APIs and will be tested inside e2e tests

**Metadata**

- https://github.com/ethereum-optimism/protocol-quest/issues/46

### CODE REVIEW

File: op-conductor/consensus/raft.go [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+		rc.log.Error("failed to shutdown raft", "err", err)
+		return err
+	}
+	return nil
+}
+
+// CommitUnsafePayload implements Consensus, it commits latest unsafe payload to the cluster FSM.
+func (rc *RaftConsensus) CommitUnsafePayload(payload eth.ExecutionPayload) error {
+	blockVersion := eth.BlockV1
+	if rc.rollupCfg.IsCanyon(uint64(payload.Timestamp)) {
```
Comment (mslipper) [inline op-conductor/consensus/raft.go:201]: Is it strictly necessary to make `op-conductor` aware of hardforks? Wouldn't it be cleaner to treat the unsafe heads data coming from Geth as a black box?

**Review by mslipper — CHANGES_REQUESTED** [review]

File: op-conductor/consensus/raft.go (hunk 1 above)
Comment (0x00101010) [reply op-conductor/consensus/raft.go:201]: Yes, it's necessary, in order to start sequencer properly at the consensus block, we need:
1. know the block number and hash to make sure sequencer has the latest block
2. and if it does not, need to gossip the block to op-node from consensus, and we need to know if we should publish to `blocksV1` or `blocksV2` topic

Comment (mslipper) [reply op-conductor/consensus/raft.go:201]: got it

Comment (mslipper) [reply op-conductor/consensus/raft.go:201]: Dumb question: can we remove this code once canyon has passed on mainnet?

**Review by mslipper — APPROVED** [review]

File: op-conductor/consensus/raft.go (hunk 1 above)
Comment (0x00101010) [reply op-conductor/consensus/raft.go:201]: Definitely can, that can be a broader cleanup of anything related to `BlockV1`

**Pull request merged**